cpp/oneapi/dal/algo/pca/common.cpp

```diff
@@ -19,49 +19,62 @@
 
 namespace oneapi::dal::pca {
 
-class detail::descriptor_impl : public base {
+template <>
+class detail::descriptor_impl<task::dim_reduction> : public base {
 public:
     std::int64_t component_count = -1;
     bool is_deterministic = false;
 };
 
-class detail::model_impl : public base {
+template <>
+class detail::model_impl<task::dim_reduction> : public base {
 public:
     table eigenvectors;
 };
 
 using detail::descriptor_impl;
```

Comment:
Is it necessary to expose internal implementation in public namespace?

Reply:
discussed offline.